cmd/refactor.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/format"
+	fileurl "github.com/open-policy-agent/opa/internal/file/url"
+	"github.com/open-policy-agent/opa/internal/refactor"
+	"github.com/open-policy-agent/opa/loader"
+)
+
+type moveCommandParams struct {
+	mapping   repeatedStringFlag
+	ignore    []string
+	overwrite bool
+}
+
+func init() {
+
+	var moveCommandParams moveCommandParams
+
+	var refactorCommand = &cobra.Command{
+		Use:    "refactor",
+		Short:  "Refactor Rego file(s)",
+		Long:   "Refactor Rego file(s).",
+		Hidden: true,
+	}
+
+	var moveCommand = &cobra.Command{
+		Use:   "move [file-path [...]]",
+		Short: "Rename packages and their references in Rego file(s)",
+		Long: `Rename packages and their references in Rego file(s).
+
+The 'move' command takes one or more Rego source file(s) and rewrites package paths and other references in them as per
+the mapping defined by the '-p' option. At least one mapping should be provided and should be of the form:
+
+	<from>:<to>
+
+The 'move' command formats the Rego modules after renaming packages, etc. and prints the formatted modules to stdout by default.
+If the '-w' option is supplied, the 'move' command will overwrite the source file instead.
+
+Example:
+--------
+
+"policy.rego" contains the below policy:
+ _ _ _ _ _ _ _ _ _ _ _ _ _
+| package lib.foo         |
+|                         |
+| default allow = false   |
+| _ _ _ _ _ _ _ _ _ _ _ _ |     
+	
+	$ opa refactor move -p data.lib.foo:data.baz.bar policy.rego
+
+The 'move' command outputs the below policy to stdout with the package name rewritten as per the mapping:
+
+ _ _ _ _ _ _ _ _ _ _ _ _ _
+| package baz.bar         |
+|                         |
+| default allow = false   |
+| _ _ _ _ _ _ _ _ _ _ _ _ | 
+`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return validateMoveArgs(args)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := doMove(moveCommandParams, args, os.Stdout); err != nil {
+				fmt.Fprintln(os.Stderr, "error:", err)
+				os.Exit(1)
+			}
+		},
```

Comment:
Isn't this approximately what happens if we use `RunE` instead?

```suggestion
		RunE: func(cmd *cobra.Command, args []string) error {
			return doMove(moveCommandParams, args, os.Stdout
		},
```
(sorry, untested) -- skimming the cobra code once more, I'm not entirely sure. It's fine, let's keep this.